### DISCUSSION TITLE
robots.txt block crawl from my components

### DISCUSSION THREAD

**Discussion by MrCasquette** [directus/directus#16698] [Archive]
As said in the title the default robots.txt block the google crawler getting info in my Navbar and footer component in NextJS. On pages the build solve the problem. But how far I know it's impossible to getStaticProps on components in NextJs and so build informations. That's why I need to have still no indexation of my api, but authorize crawling. where's the robot or where to config?
I use a self hosted, so i found (I think) the public folder in /@directus/app/dist but do I have to put a robot here or ... ? I don't know
Thank for any reply

**Comment by br41nslug** [comment]
Extending the `robots.txt` is an open issue im afraid https://github.com/directus/directus/issues/15421 however there is a workaround extension in the issue comments.

**Reply by MrCasquette** [reply]
Thanks for the reply. That's what I was afraid of too 😅 I'll probably try to make my first extension. Thank you

**Reply by br41nslug** [reply]
The example that is in the issue should work with just copy-pasting it into your local `extensions/hooks/extension-name/index.js` folder 😄

**Reply by MrCasquette** [reply]
It doesn't  work, i create an extension in the extension folder, edit the index.js with the pasted value
`const customRobotsTxt = `
User-agent: *
Allow: /
`.trim();

module.exports = ({ init }) => {
  init("app.before", ({ app }) => {
    app.get("/robots.txt", (_, res) => {
      res.set("Content-Type", "text/plain");
      res.status(200);
      res.send(customRobotsTxt);
    });
  });
};`

I build the extension, restart my app, goes to /robots.txt it's still disallow

**Reply by br41nslug** [reply]
no need to build just copy-paste the index.js and restart, and make sure you're using the right path.

**Reply by MrCasquette** [reply]
It's already did. Maybe it's in the wrong place. What the exact path where I have to put it?

**Reply by br41nslug** [reply]
`extensions/hooks/extension-name/index.js` where extensions is the `extensions` folder configured in directus (in your install root for local npm deployments, for docker you'll have to add a volume).

**Reply by MrCasquette** [reply]
The extension i build is exactly here : `/mySite/mysite-directus/extensions/hooks/robot_maker/src/index.js`
do i move up index of src or maybe rename the robot_maker folder?

**Reply by br41nslug** [reply]
Yep move it up one, the `src` is only used in the source bootstrap of extensions.

**Reply by MrCasquette** [reply]
So simple and yet... you need to know 😅 It works perfectly thanks.

**Reply by MrCasquette** [reply]
Maybe another thing how to disable index now?

**Reply by br41nslug** [reply]
Indexing is disabled by default you'll want to take that robots.txt and allow the bots you want to be able to index :)

**Reply by MrCasquette** [reply]
I read somewhere robot is only for crawling not for index. Disable if crawling is disable obviously. But I want google to be permitted to crawl and be able to get data from my front, but not to be able to index directus obviously too. Is there a way? Because my robot is now user-agent * Allow /

**Reply by br41nslug** [reply]
I am not super familiar with the specific syntax of robots.txt as i tend to just block everthing 😬 Imo crawling and indexing are 2 sides of the same coin where crawling is the act of moving through all reachable pages and indexing is saving the resulting data. What specifically do you want to allow and what do you want to block?

**Reply by MrCasquette** [reply]
Actually I would have preferred to let the default disallow. But as I said I see on google search console that my networks, and contact info stored in directus and reach via react on my nextJS app can't be accessed by google. Every other thing are build with getStaticProps method of next but as my navbar and footer are component I cannot use this method. So I would want these informations to be visible for google but my directus not indexed. I hope I'm clear.

**Reply by MrCasquette** [reply]
Forget about it, I've done it another way. Rule number one of the code: If it doesn't work, do it another way 😅 I get back my robot to disallow all and succeed in finding a way to build my infos and networks globally. Thank you for your time. I love you all directus Team. Great project, great member, available... Never change you are all awesome I LOVE DIRECTUS !!!! 😄
